### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by stuartgo**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ dependencies = [
     "uvicorn>=0.19.0",
     "cloudpickle>=2.2.0",
     "mlflow>=1.30.0",
+    "sklearn>=0.0",
```
Comment (georgedouzas) [inline pyproject.toml:33]: You mean `scikit-learn` dependency?

File: tests/services/service5/test_service5.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+client = TestClient(app)
+
+def test_get_model():
+
+    response = client.post("http://localhost:8001/get_model",json={"model_name":"sklearn_vectorizer"})
+    model=cloudpickle.loads(base64.b64decode(response.json()["model"]))
+    assert isinstance(model,mlflow.pyfunc.PyFuncModel)
+    assert response.status_code == 200
+
+def test_avaliable_models():
```
Comment (georgedouzas) [inline tests/services/service5/test_service5.py:19]: Typo

Comment (georgedouzas) [inline tests/services/service5/test_service5.py:19]: Typo

File: tests/services/service1/test_service1.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+client = TestClient(app)
+
+def test_model():
+    vectorizer=Vectorizer()
+    vectorizer.fit(["test 123, this is a test sentence","Barack Obama was the president"])
+    model_info = mlflow.sklearn.log_model(sk_model=vectorizer, artifact_path="model")
+    sklearn_pyfunc = mlflow.pyfunc.load_model(model_uri=model_info.model_uri)
+    response = client.post("http://localhost:8000/model",json={"model":base64.b64encode(cloudpickle.dumps(sklearn_pyfunc)).decode()})
+    assert response.status_code == 200
```
Comment (georgedouzas) [inline tests/services/service1/test_service1.py:22]: How do we test if the model is actually used?

File: tests/services/service1/test_service1.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+def test_vectorisation():
+    docs=["this is a test 123"]
+    response = client.post("http://localhost:8000/vectorisation",json={"docs":docs})
+    assert response.status_code == 200
+    assert isinstance(response.json()["embeddings"],list)
+    assert isinstance(response.json()["embeddings"][0],list)
+    assert isinstance(response.json()["embeddings"][0][0],(int,float))
+    assert len(docs)==len(response.json()["embeddings"])
+
```
Comment (georgedouzas) [inline tests/services/service1/test_service1.py:32]: I guess here the service is responding with a json similar to the one used in the current system? (dev branch)

File: tests/services/service5/test_service5.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    response = client.post("http://localhost:8001/available_models",json={})
+    assert isinstance(response.json()["models"][0],str)
+    assert response.status_code == 200
+
```
Comment (georgedouzas) [inline tests/services/service5/test_service5.py:24]: We need to test the actual names.

File: tests/services/service1/test_service1.py (hunk 3 above)
Comment (stuartgo) [reply tests/services/service1/test_service1.py:22]: What do you mean? If the model sent is the one being stored?

Comment (georgedouzas) [reply tests/services/service1/test_service1.py:22]: This endpoint is used to update the model, correct? How do we know it actually updated the model?

**Pull request merged**